packages/eslint-plugin/src/rules/strict-boolean-expressions.ts

```diff
@@ -909,7 +910,11 @@ export default createRule<Options, MessageId>({
             ),
         )
       ) {
-        variantTypes.add('object');
+        if (types.some(type => isBrandedBoolean(type))) {
+          variantTypes.add('branded boolean');
```

Comment:
[Refactor] Also, I switched this to be `boolean`:

```suggestion
          variantTypes.add('boolean');
```

...and all the current tests pass. Which I think kind of makes sense. I can't think of any reason why the rule would care about the difference between a _branded_ boolean and a non-branded one. Can you?

* If there are reasons to care, they should be in the tests.
* If not, let's stick with just `'boolean'`